jcli/build.rs

```diff
@@ -0,0 +1,82 @@
+use std::env::consts::{ARCH, OS};
+use std::process::Command;
+
+#[cfg(debug_assertions)]
+const BUILD_TYPE: &'static str = "debug";
+#[cfg(not(debug_assertions))]
+const BUILD_TYPE: &'static str = "release";
+
+fn main() {
+    let version_string = format!(
+        "{} ({}-{}{}, {}, {} [{}]) - [{}]",
```

Comment:
This would be easier to understand if it used named format patterns.

Reply:
is this ok?
```rust
    let version_string = format!(
        "{pkg_version} ({branch_name}-{commit_hash}{dirty}, {build_type}, {os_type} [{cpu_arch}]) - [{rustc_version}]",
        pkg_version = env!("CARGO_PKG_VERSION"),
        branch_name = git_branch_name(),
        commit_hash = git_commit_hash(),
        dirty = if git_working_tree_is_clean() { "" } else { "+" },
        build_type = BUILD_TYPE,
        os_type = OS,
        cpu_arch = ARCH,
        rustc_version = compiler_version()
    );
```